test/macros.jl

```diff
@@ -176,6 +183,75 @@ function build_constraint_keyword_test(ModelType::Type{<:JuMP.AbstractModel})
     end
 end
 
+struct MyConstrType end
+struct BadPosArg end
+function JuMP.build_constraint(
+    _error::Function,
+    f::AffExpr,
+    set::MOI.EqualTo,
+    extra::Type{MyConstrType};
+    d = 0,
+)
+    new_set = MOI.LessThan{Float64}(set.value + d)
+    return JuMP.build_constraint(_error, f, new_set)
+end
+function build_constraint_extra_arg_test(ModelType::Type{<:JuMP.AbstractModel})
```

Comment:
You need to call this function to test it.

Reply:
Done!